test/system/110-editor-login.js

```diff
@@ -0,0 +1,233 @@
+        params.redirect_uri = redirectCallback
+        const response = await app.inject({
+            method: 'GET',
+            url: `/account/authorize?${new URLSearchParams(params)}`,
+            headers: {
+                authorization: `Bearer ${userToken}`
+            }
+        })
+        response.statusCode.should.equal(302)
+        response.headers.should.have.property('location')
+
+        // http://localhost:3000/account/request/z7TucmJ-avnR1eyJLUII5wccsNn3EiU3CpBbpdxx9wg/editor
+
+        const parts = response.headers.location.split('/')
+        const token = parts[parts.length - 2]
+
+        const response2 = await app.inject({
+            method: 'GET',
+            url: `/account/complete/${token}`,
+            cookies: {
+                sid: userToken
+            }
+        })
+
+        if (dashboardOnly) {
+            response2.statusCode.should.equal(400)
+            response2.body.should.equal('Access Denied: you do not have access to the editor')
+            return
+        }
+        response2.statusCode.should.equal(302)
+        response2.headers.should.have.property('location')
+        // http://example.com/flowforge-nr-tools/auth/callback?code=eANs21GUv7OqN99S2QxJ4tS1BD5RYEzOEfb-lhLUciw&state=YMbFnV1E_2aPwz_Ktca1DQ
+        const callbackURL = new URL(response2.headers.location)
+        callbackURL.pathname.should.equal('/auth/callback')
+        callbackURL.host.should.equal('example.com')
+        should.exist(callbackURL.searchParams.get('code'))
+        should.exist(callbackURL.searchParams.get('state'))
+        callbackURL.searchParams.get('state').should.equal(state)
+
+        const params2 = {}
+        params2.grant_type = 'authorization_code'
+        params2.code = callbackURL.searchParams.get('code')
+        params2.redirect_uri = redirectCallback
+        params2.client_id = TestObjects.tokens.Instance1.clientID
+        params2.code_verifier = verifier
+
+        const response3 = await app.inject({
+            url: '/account/token',
+            method: 'POST',
+            payload: params2,
+            headers: {
+                authorization: `Bearer ${userToken}`
+            }
+        })
+        response3.statusCode.should.equal(200)
+        const body = JSON.parse(response3.body)
+        body.should.have.property('state', state)
+        return body
+    }
+
+    it('editor login via oauth flow - full access', async function () {
+        const tokens = await doEditorLogin(TestObjects.tokens.alice, 'editor-0.18', false)
+        tokens.should.have.property('scope', '*')
+        const response = await app.inject({
+            url: '/api/v1/user',
+            method: 'GET',
+            headers: {
+                authorization: `Bearer ${tokens.access_token}`
+            }
+        })
+        response.statusCode.should.equal(200)
+        const result = JSON.parse(response.body)
+        result.should.have.property('username', 'alice')
+    })
```

Comment:
[nb] the inconsistent spacing in this file is driving me a little :chestnut:'s 

```suggestion
    })

```